lib/Common/Memory/PageAllocator.cpp

```diff
@@ -977,29 +977,18 @@ PageAllocatorBase<TVirtualAlloc, TSegment, TPageSegment>::FillAllocPages(__in vo
 
 #if DBG
 #ifdef RECYCLER_ZERO_MEM_CHECK
-    const bool isLocalProc = this->processHandle == GetCurrentProcess();
-    byte * readBuffer;
-    if (isLocalProc)
+    byte * localAddr = (byte *)this->GetVirtualAllocator()->AllocLocal(address, bufferSize);
+    if (!localAddr)
     {
-        readBuffer = (byte*)address;
-    }
-    else
-    {
-        readBuffer = HeapNewArray(byte, bufferSize);
-        if (!ReadProcessMemory(this->processHandle, address, readBuffer, bufferSize, NULL))
-        {
-            MemoryOperationLastError::RecordLastErrorAndThrow();
-        }
+        MemoryOperationLastError::RecordError(E_OUTOFMEMORY);
+        return;
     }
     for (size_t i = 0; i < bufferSize; i++)
     {
         // new pages are filled with zeros, old pages are filled with DbgMemFill
-        Assert(readBuffer[i] == 0 || readBuffer[i] == DbgMemFill);
-    }
-    if (!isLocalProc)
-    {
-        HeapDeleteArray(bufferSize, readBuffer);
+        Assert(localAddr[i] == 0 || localAddr[i] == DbgMemFill);
     }
+    this->GetVirtualAllocator()->FreeLocal(localAddr);
```

Comment:
for in proc, yes, for oop it unmaps the local view

---
In reply to: [104337005](https://github.com/Microsoft/ChakraCore/pull/2619#discussion_r104337005) [](ancestors = 104337005)